rules/typescript/original.js

```diff
@@ -0,0 +1,457 @@
+module.exports = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: `${process.cwd()}/tsconfig.json`,
+  },
+  plugins: ['@typescript-eslint'],
+  rules: {
+    // Require that member overloads be consecutive
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/adjacent-overload-signatures.md
+    '@typescript-eslint/adjacent-overload-signatures': 'error',
+
+    // Requires using either T[] or Array<T> for arrays
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/array-type.md
+    '@typescript-eslint/array-type': [
+      'error',
+      {
+        default: 'array-simple',
+        readonly: 'array-simple',
+      },
+    ],
+
+    // Disallows awaiting a value that is not a Promise
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/await-thenable.md
+    '@typescript-eslint/await-thenable': 'error',
+
+    // Bans // @ts-<directive> comments from being used or requires descriptions
+    // after directive
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-ts-comment.md
+    // EXPLANATION: Complete banning is too strict, but requiring the
+    // description is a nice idea
+    '@typescript-eslint/ban-ts-comment': [
+      'error',
+      {
+        'ts-expect-error': 'allow-with-description',
+        'ts-ignore': 'allow-with-description',
+        'ts-nocheck': 'allow-with-description',
+        'ts-check': 'allow-with-description',
+        minimumDescriptionLength: 3,
+      },
+    ],
+
+    // Bans // tslint:<rule-flag> comments from being used.
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-tslint-comment.md
+    '@typescript-eslint/ban-tslint-comment': 'error',
+
+    // Bans specific types from being used
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md
+    // EXPLANATION: No specific types to ban
+    '@typescript-eslint/ban-types': 'off',
+
+    // Ensures that literals on classes are exposed in a consistent style
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/class-literal-property-style.md
+    '@typescript-eslint/class-literal-property-style': ['error', 'fields'],
```

Comment:
This rule only works well for `.ts` files. For `.js` files, it suggests adding `readonly` which is not a valid JS syntax.

<img width="1035" alt="image" src="https://user-images.githubusercontent.com/5039436/176368694-3a018ac6-df95-46e9-a026-9c43409b64c0.png">

Reply:
I guess there is something went wrong with distinguishing `.ts` from `.js` files. I'll take a look.

Reply:
According to the internal discussion, we decided to keep this rule as is